library/btreeRecurssion.js

```diff
@@ -0,0 +1,63 @@
+"use strict";
+
+class Node {
+  /**
+   * creating a node for binary tree.
+   * @param {any} data
+   * @return {node}
+   */
+  constructor(data) {
+    this.right = null;
+    this.data = data;
+    this.left = null;
+  }
+
+  addLink(nextNode, data) {
+    /**
+     * Searching through the tree till a nul space is found and adding a new node at that position.
+     * @param {any} data
+     * @return {node}
+     */
+    if (!nextNode.data) {
+      nextNode.data = data;
+      return this;
+    }
+    if (nextNode.data > data && nextNode.left === null) {
+      nextNode.left = new Node(data);
+      return this;
+    } else if (nextNode.data < data && nextNode.right === null) {
+      nextNode.right = new Node(data);
+      return this;
+    } else if (nextNode.data > data && nextNode.left != null) {
+      nextNode = nextNode.left;
+      return this.addLink(nextNode, data);
+    } else {
+      nextNode = nextNode.right;
+      return this.addLink(nextNode, data);
+    }
+  }
+
+  searchTree(nextNode, data) {
+    /**
+     * Searching through the tree to find the value.
+     * @param {any} data
+     * @return {string}
+     */
+    if (nextNode.data === data) {
+      return "value exists";
+    }
+    if (nextNode.data > data && nextNode.left != null) {
```

Comment:
in this case, a little easier to read if you made the ifs
```
if < data: 
  if left not null
   else
elif > data
```

Reply:
adjusted.